### DISCUSSION TITLE
UNABLE TO IMPORT SNOMED AFTER INSTALL

### DISCUSSION THREAD

**Discussion by okongwu101** [b2ihealthcare/snow-owl#940] [Q&A] [answered]
**Hi. So i installed snow-owl using the docker image. The installation appear to be working as i entered this in the prompt:**

`curl http://localhost:8080/snowowl/admin/info
`

**and got the following response:**

`{"version":"7.17.3","description":"You Know, for Terminologies","repositories":{"items":[{"id":"snomedStore","health":"GREEN","diagnosis":"","indices":[{"index":"snomedstore-branch","status":"GREEN"},{"index":"snomedstore-review","status":"GREEN"},{"index":"snomedstore-concretedomainchange","status":"GREEN"},{"index":"snomedstore-conceptchange","status":"GREEN"},{"index":"snomedstore-concept","status":"GREEN"},{"index":"snomedstore-equivalentconceptset","status":"GREEN"},{"index":"snomedstore-commit","status":"GREEN"},{"index":"snomedstore-codesystemversion","status":"GREEN"},{"index":"snomedstore-classificationtask","status":"GREEN"},{"index":"snomedstore-conceptchanges","status":"GREEN"},{"index":"snomedstore-constraint","status":"GREEN"},{"index":"snomedstore-relationshipchange","status":"GREEN"},{"index":"snomedstore-description","status":"GREEN"},{"index":"snomedstore-relationship","status":"GREEN"},{"index":"snomedstore-member","status":"GREEN"},{"index":"snomedstore-codesystem","status":"GREEN"},{"index":"snomedstore-descriptionchange","status":"GREEN"}]}]},"cluster":{"available":true,"diagnosis":"","healthy":true,"indices":[{"index":"snomedids-sctid","status":"GREEN"},{"index":"locks-lock","status":"GREEN"},{"index":"validations-validationrule","status":"GREEN"},{"index":"jobs-job","status":"GREEN"},{"index":"validations-validationwhitelist","status":"GREEN"},{"index":"validations-validationissue","status":"GREEN"}]}}` .

**However when i try to upload snomed using:**

`curl -X POST http://localhost:8080/snowowl/snomed-ct/v3/imports -d
{
  "type": "SNAPSHOT",
  "branchPath": "MAIN"
}
`

**i get the message**

`{"status":415,"code":0,"message":"HTTP Media Type application/x-www-form-urlencoded is not supported. Supported media types are: [application/json;charset=UTF-8]","errorCode":0,"statusCode":415}curl: (6) Could not resolve host: type
curl: (6) Could not resolve host: SNAPSHOT,
curl: (6) Could not resolve host: branchPath
curl: (3) [globbing] unmatched close brace/bracket in column 5`

** Any help with this is highly appreciated. Thanks.**

**Comment by cmark** [comment, marked as answer]
Hi @okongwu101,

Thank you for checking out Snow Owl.
It looks like the `curl` command you have tried missing a few required parameters to work properly. 
I recommend using the newer `POST /snomed-ct/v3/{path}/import` endpoint instead of the deprecated `POST /snomed-ct/v3/imports` (the latter will be removed in the upcoming 8.x release).

An example curl command to import the latest International RF2 Full from the official release file would look like this:
```
curl -X POST "http://localhost:8080/snowowl/snomed-ct/v3/MAIN/import?type=Full&createVersions=true" -H "Content-Type: multipart/form-data" -F "file=@/path/to/SnomedCT_InternationalRF2_PRODUCTION_20210731T120000Z.zip" 
```

I hope this helps.

Regards,
Mark

**Comment by okongwu101** [comment]
Hi @cmark ,
Thank you for the quick response. Will try it and get back to you.

**Comment by okongwu101** [comment]
`{"status":401,"code":0,"message":"Missing authorization token","developerMessage":"Missing authorization token","statusCode":401,"errorCode":0}`

That was the error message i got. How do i acquire the authorization code? Any guide on how to use it? Thanks in advance.

**Comment by cmark** [comment]
Did you set up any form of authorization? If no then the default file-based authorization is in place with a single default user `username: snowowl, pass: snowowl`. (for production deploys please use a strong password or configure LDAP-based authorization).
You can use the Base64 encoded form of the `username:password` string in a header when you send any HTTP request to Snow Owl: 
```
curl .... -H "Authorization: Basic c25vd293bDpzbm93b3ds" ...
```

I hope this helps. Documentation can be found [here](https://docs.b2i.sg/snow-owl/), how ever regarding API usage we don't have a detailed documentation publicly available yet (it is expected to come in the coming weeks though)

**Comment by okongwu101** [comment]
Worked ! Thanks a lot. Up to the next brick wall.

**Comment by cmark** [comment]
Hi @okongwu101,

Do you still experience issues with your RF2 import? Have you been able to import any RF2 content successfully into your installation?

Cheers,
Mark

**Comment by cmark** [comment]
_Converting to discussion..._
